python/tests/stress/arcticdb/test_mem_leaks.py

```diff
@@ -0,0 +1,201 @@
+        of memory since iterations started as weel as minumum growth of memory. That along with information
+        how many times the growth of mem droped 50% could provide insight for presence or lack of memory issues
+
+        NOTE: This test is NOT CONCLUSIVE for absence of memory leacks. It would catch LARGE leaks. 
+        Small leaks will not be caught. If the procedure fails to sport problems it will return 2 numbers
+        for further investigation by the test:
+         - how much the process has grown during execution in bytes
+         - how many times the process memory dropped more than 50% (indicating some C++ garbage collection)
+
+        NOTE 2: Currently this code controlls python GC only.
+
+    """
+    p = get_process_by_id(os.getpid())
+    avail_mem_start : np.int64 = psutil.virtual_memory().available
+    print ("-" * 80)
+    print ("Start check process for memory leaks")
+    print ("Num iterations: ", number_iterations)
+    print ("Maxumum memory growth/lost to the examined process: ", max_total_mem_lost_treshold)
+    print ("Maxumum machine memory utilization: ", max_total_mem_lost_treshold)
+    
+    lets_collect_some_garbage(10)
+
+    mem_per : np.int64 = psutil.virtual_memory().percent
+    assert max_machine_memory_percentage >- mem_per , "Machine has less memory available the check cannot start"    
+    print ("Available memory" , nice_bytes_str(avail_mem_start))
+    print ("-" * 80)
+    process_initial_memory : np.int64 = p.memory_info().rss
+    print ("Process initial RSS", nice_bytes_str(process_initial_memory))
+
+    mem_growth_each_iter = list()
+
+    for n in range(number_iterations):
+        p_iter_mem_start : np.int64  = p.memory_info().rss
+        
+        print("Starting watched code ...........")
+        process_func()
+        lets_collect_some_garbage()
+
+        p_iter_mem_end : np.int64  = p.memory_info().rss
+        process_growth : np.int64 = p.memory_info().rss - process_initial_memory
+        mem_growth_each_iter.append(process_growth)
+        process__avg_growth = process_growth / (n + 1)
+        p_iter_mem_notcleaned : np.int64  = p_iter_mem_end - p_iter_mem_start
+        mem_avail : np.int64 = psutil.virtual_memory().available
+        mem_per : np.int64 = psutil.virtual_memory().percent
+        print (f"Iter No[{n}]    : Process did added (or if - number means cleaned) {nice_bytes_str(p_iter_mem_notcleaned)}.  Avail memory: {nice_bytes_str(mem_avail)} Used Mem: {mem_per}%")
+        print (f"  Overall stats : Process growth since start {nice_bytes_str(process_growth)} AVG growth per iter {nice_bytes_str(process__avg_growth)}")
+        print (f"  Minimum growth so far: {nice_bytes_str(min(mem_growth_each_iter))}")
+        print (f"  Maximum growth so far: {nice_bytes_str(max(mem_growth_each_iter))}")
+        print (f"  Number of times there was 50% drop in memory: {count_drops(mem_growth_each_iter, 0.5)}")
```

Comment:
How do you think people will use the stats about the number of drops?

Reply:
The number of drops means there is garbage collection going on. Thus you might want to extend with more repeats if garbage collection goes but it is not enough ... Potentially this metric is here to get you oriented what is going on during execution. Thus finetuning of this is important by running pytest with -s and only this test from command line. I will create also a wiki page with advises if needed